spec/services/work_combined_audio_creator_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+describe "Combined Audio" do
+  # We need to set the friendlier_id explicitly, because it's part of the fingerprint recipe.
+  let!(:work) { FactoryBot.create(:work, title: "top", friendlier_id: '1ciyp1i')}
+
+  let!(:audio_asset_1)  { create(:asset, :inline_promoted_file,
+      position: 1,
+      parent_id: work.id,
+      id: 'ddbd1a8d-c2eb-47b3-85a4-11b4ffb41719',
+      file: File.open((Rails.root + "spec/test_support/audio/ice_cubes.mp3"))
+    )
+  }
+
+  let!(:audio_asset_2)  { create(:asset, :inline_promoted_file,
+      position: 2,
+      parent_id: work.id,
+      id: 'df773502-56d7-4756-a58c-b1f479910e97',
+      file: File.open((Rails.root + "spec/test_support/audio/double_ice_cubes.mp3"))
+    )
+  }
+
+  it "creates combined audio derivatives" do
+    expect(work.members.map(&:stored?)).to match([true, true])
+    combined_audio_info = CombinedAudioDerivativeCreator.new(work).generate
+    expect(combined_audio_info.start_times.count).to eq 2
+    expect(combined_audio_info.start_times).to match([
+      ["ddbd1a8d-c2eb-47b3-85a4-11b4ffb41719", 0],
+      ["df773502-56d7-4756-a58c-b1f479910e97", 1.593469]
+    ])
+    cmd = TTY::Command.new(printer: :null)
+    stats_command = ['ffprobe', '-v', 'error', '-show_format', '-show_streams' ]
+    mp3_details   = cmd.run(*stats_command + [combined_audio_info.mp3_file.path] ).out.split("\n")
+    webm_details  = cmd.run(*stats_command + [combined_audio_info.webm_file.path]).out.split("\n")
+    expect(mp3_details).to  include('format_name=mp3')
+    expect(webm_details).to include('format_name=matroska,webm')
+
+    expect(mp3_details.any?  {|x| x.include? 'duration=4.7'}).to be true
+    expect(webm_details.any? {|x| x.include? 'duration=4.7'}).to be true
+    # The fingerprint depends on the title, friendlier_id, and audio_file_sha512.
+    # As long as those are fixed, you can count on this fingerprint.
+    expect(combined_audio_info.fingerprint).to eq '8402b003dd07775a83b1a000b3319fd9'
```

Comment:
Prob just necessar to check that `fingerprint` is present?, rather than that it's a specific thing, we don't really know what it's supposed to be, don't need the test to break if we change algorithm, but not really a big deal either way.

Reply:
Cool!